Services/WebServices/ECS/classes/Consent/class.ilECSUserConsent.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ */
+
+/**
+ * Class ilECSUserConsent
+ * @author Stefan Meyer <smeyer.ilias@gmx.de>
+ */
+class ilECSUserConsent
+{
+    private int $usr_id;
+    private int $mid;
+
+    protected ilDBInterface $db;
+
+    public function __construct(int $a_usr_id, int $a_mid)
+    {
+        global $DIC;
+
+        $this->db = $DIC->database();
+
+        $this->usr_id = $a_usr_id;
+        $this->mid = $a_mid;
+
+    }
+
+    public function getUserId() : int
+    {
+        return $this->usr_id;
+    }
+
+    public function getMid() : int
+    {
+        return $this->mid;
+    }
+
+    public function save() : void
+    {
+        try {
+            $query = 'INSERT INTO ecs_user_consent (usr_id, mid) ' .
+                'VALUES ( ' .
+                $this->db->quote(
+                    $this->getUserId(), ilDBConstants::T_INTEGER
+                ) . ', ' .
+                $this->db->quote(
+                    $this->getMid(), ilDBConstants::T_INTEGER
+                ) . ' ' .
+                ')';
+            $this->db->manipulate($query);
+        } catch (ilDatabaseException $e) {
+            // ignore duplicate entry
+        }
+    }
+
+    public function delete() : void
+    {
+        $query = 'DELETE FROM ecs_user_consent ' .
+            'WHERE usr_id = ' . $this->db->quote(
+                $this->getUserId(), ilDBConstants::T_INTEGER
+            ) . ' ' .
+            'AND mid = ' . $this->db->quote(
+                $this->getMid(), ilDBConstants::T_INTEGER
+            );
+        $this->db->manipulate($query);
+    }
+}
```

Comment:
This is a coding standard violation.

I strongly recommend using `CaptainHook` or a respective inspection profile in PhpStorm to detect and fix such issues.